src/ApolloProvider.tsx

```diff
@@ -29,17 +32,23 @@ export default class ApolloProvider<TCache> extends Component<ApolloProviderProp
       'ApolloClient was not passed a client instance. Make ' +
         'sure you pass in your client via the "client" prop.',
     );
+
+    // we have to attach to the client since you could have multiple
+    // providers
+    // XXX this is backwards compat and will be removed in 3.0
+    if (!(props.client as any).__operations_cache__) {
+      (props.client as any).__operations_cache__ = this.operations;
+    }
   }
 
   getChildContext() {
     return {
       client: this.props.client,
+      operations: (this.props.client as any).__operations_cache__,
     };
   }
 
   render() {
-    return (
-      <QueryRecyclerProvider>{React.Children.only(this.props.children)}</QueryRecyclerProvider>
-    );
+    return React.Children.only(this.props.children);
```

Comment:
I'm not sure we care about children being one and only one here.  Any reason (with react 16) that we shouldn't just `return this.props.children`?  If that complains, then just `return <React.Fragment>{this.props.children}</React.Fragment`

Reply:
ah yeah that sounds great to me